client/my-sites/domains/domain-management/components/primary-domain-selector/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { FEATURE_SET_PRIMARY_CUSTOM_DOMAIN } from '@automattic/calypso-products';
+import { useTranslate } from 'i18n-calypso';
+import { useState, ChangeEvent, useEffect } from 'react';
+import FormSelect from 'calypso/components/forms/form-select';
+import type { DomainData, SiteDetails } from '@automattic/data-stores';
+
+import './style.scss';
+
+type PrimaryDomainSelectorProps = {
+	domains: undefined | DomainData[];
+	userCanSetPrimaryDomains: boolean;
+	site: undefined | null | SiteDetails;
+	onSetPrimaryDomain: ( domain: string, onComplete: () => void ) => void;
+};
+
+const PrimaryDomainSelector = ( {
+	domains,
+	site,
+	userCanSetPrimaryDomains,
+	onSetPrimaryDomain,
+}: PrimaryDomainSelectorProps ) => {
+	const [ selectedDomain, setSelectedDomain ] = useState< undefined | string >( undefined );
+	const [ isSettingPrimaryDomain, setIsSettingPrimaryDomain ] = useState< boolean >( false );
+
+	const translate = useTranslate();
+
+	useEffect( () => {
+		if ( domains?.length ) {
+			const primaryDomain = domains.find( ( domain ) => {
+				return domain.primary_domain;
+			} );
+			if ( primaryDomain ) {
+				setSelectedDomain( primaryDomain.domain );
+			}
+		}
+	}, [ domains ] );
+
+	if ( ! domains || ! site ) {
+		return null;
+	}
+
+	const isOnFreePlan = site?.plan?.is_free ?? false;
+
+	const shouldUpgradeToMakeDomainPrimary = ( domain: DomainData ): boolean => {
+		return (
+			! userCanSetPrimaryDomains &&
+			isOnFreePlan &&
+			( domain.type === 'registered' || domain.type === 'mapping' ) &&
+			! domain.current_user_can_create_site_from_domain_only &&
+			! domain.wpcom_domain &&
+			! domain.is_wpcom_staging_domain &&
+			( site?.plan?.features.active.includes( FEATURE_SET_PRIMARY_CUSTOM_DOMAIN ) ?? false )
+		);
+	};
+
+	const validPrimaryDomains = domains.filter( ( domain ) => {
+		return (
+			domain.can_set_as_primary &&
+			! domain.aftermarket_auction &&
+			domain.points_to_wpcom &&
+			! shouldUpgradeToMakeDomainPrimary( domain )
+		);
+	} );
+
+	const onSelectChange = ( event: ChangeEvent< HTMLSelectElement > ) => {
+		setSelectedDomain( event.target.value );
+		setIsSettingPrimaryDomain( true );
+		onSetPrimaryDomain( event.target.value, () => setIsSettingPrimaryDomain( false ) );
+	};
+
+	return (
+		<div className="domains-set-primary-address">
+			<div className="domains-set-primary-address__title">
+				{ translate( 'Primary site address' ) }
+			</div>
+			<div className="domains-set-primary-address__subtitle">
+				{ translate( 'The current primary address set for this site is:' ) }{ ' ' }
```

Comment:
What do you think about using the `InlineSupportLink` component to add a "Learn more" link to this guide? https://wordpress.com/support/domains/set-a-primary-address/

I'm not sure if users will be familiar with what a "Primary site address" actually means for them, and the effect that it will have on their site.

Reply:
Nice! I was considering to update the support page before adding any link but you're right, pointing to set primary address page it's a good idea.